discovery/azure/azure.go

```diff
@@ -212,22 +237,26 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 		return tg, fmt.Errorf("could not create Azure client: %s", err)
 	}
 
-	var machines []compute.VirtualMachine
-	result, err := client.vm.ListAll()
+	machines, err := client.getVMs()
 	if err != nil {
-		return tg, fmt.Errorf("could not list virtual machines: %s", err)
+		return tg, fmt.Errorf("could not get virtual machines: %s", err)
 	}
-	machines = append(machines, *result.Value...)
 
-	// If we still have results, keep going until we have no more.
-	for result.NextLink != nil {
-		result, err = client.vm.ListAllNextResults(result)
+	level.Debug(d.logger).Log("Found %d virtual machines during Azure discovery.", len(machines))
```

Comment:
This isn't following the usual go-kit way of doing structured logging.

Reply:
result network.Interface,